p2p/dandelion/src/traits.rs

```diff
@@ -0,0 +1,49 @@
+/// A request to diffuse a transaction to all connected peers.
+///
+/// This crate does not handle diffusion it is left to implementers.
+pub struct DiffuseRequest<Tx>(pub Tx);
+
+/// A request sent to a single peer to stem this transaction.
+pub struct StemRequest<Tx>(pub Tx);
```

Comment:
```suggestion
/// A request to diffuse a transaction to all connected peers.
///
/// This crate does not handle diffusion it is left to implementers.
#[repr(transparent)]
pub struct DiffuseRequest<Tx>(pub Tx);

/// A request sent to a single peer to stem this transaction.
#[repr(transparent)]
pub struct StemRequest<Tx>(pub Tx);
```

Reply:
Why? no one should be transmuting these, surly we should let Rust decide the layout?

Reply:
Should have noted it was a suggestion if applicable.

I'd usually follow [this](https://rust-lang.github.io/api-guidelines/interoperability.html#types-eagerly-implement-common-traits-c-common-traits) (and extend it to `repr`) if you wanted to guarantee layout for users, up to you here.